ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/bootstrap/FactoryWorkspaceComponent.java

```diff
@@ -167,20 +172,28 @@ public Boolean apply(UsersWorkspaceDto arg) throws FunctionException {
                         //TODO rework it when account will be ready
                         return workspaceConfigDto.getName().equals(arg.getName());
                     }
-                });
+                }, false);
             case "perClick":
             default:
-                return workspaceServiceClient.create(workspaceConfigDto.withName("xXx"), null);
+                return getWorkspaceByConditionOrCreateNew(workspaceConfigDto, new Function<UsersWorkspaceDto, Boolean>() {
+                    @Override
+                    public Boolean apply(UsersWorkspaceDto workspaceDto) throws FunctionException {
+                        return workspaceDto.getName().equals(workspaceConfigDto.getName());
+                    }
+                }, true);
         }
     }
 
     /**
      * Gets the workspace by condition which is determined by given {@link Function}<br/>
-     * if workspace found by condition then it will return {@link Promise} of it<br/>
-     * else it wil produce {@link Promise} of new workspace.
+     * if workspace found by condition and {@code generateName} is defined as {@code true}<br/>
+     * it create instance of workspace with generated name and return {@link Promise} of it<br/>
+     * if generateName is {@code false} it will return {@link Promise} of founded workspace,<br/>
+     * if workspace not found by condition Promise of new workspace will be returned.
      */
     private Promise<UsersWorkspaceDto> getWorkspaceByConditionOrCreateNew(final WorkspaceConfigDto workspaceConfigDto,
-                                                                          final Function<UsersWorkspaceDto, Boolean> condition) {
+                                                                          final Function<UsersWorkspaceDto, Boolean> condition,
+                                                                          final boolean generateName) {
```

Comment:
I think it would be better to name this parameter something like useExists or createNewWorkspace